### ISSUE TITLE
other authentication methods do not work

### ISSUE THREAD

**Issue by CNYuYang** [goharbor/harbor#13734]
I want to implement identity authentication in the api calls of harbor and adapter

But I used the other three authentication methods in turn, they don't seem to work

- harbor version: 2.1.1
- docker engine version: 19.03.13
- docker-compose version: 1.26.1

# Apikey

My Config:

![apikey](https://user-images.githubusercontent.com/42082796/101852790-64140200-3b99-11eb-8dec-689879b6331e.png)

Browser Console:

```json
{"errors":[{"code":"BAD_REQUEST","message":"scanner API: ping: auth type APIKey is not supported"}]}
```

# Basic

My Config:

![basic](https://user-images.githubusercontent.com/42082796/101852851-83ab2a80-3b99-11eb-93ee-fc49886d084f.png)

Browser Console:

```json
{"errors":[{"code":"UNKNOWN","message":"internal server error"}]}
```

Harbor log:

```txt
Dec 10 18:02:27 172.21.0.1 core[ ]: 2020- - T : : Z [ERROR] [/lib/http/error.go: ]: {"errors":
[{"code":"INTERNAL_SERVER_ERROR","message":"scanner API: ping: scanner controller: ping: v client:
get metadata: send: authorization: Basic: missing data to authorize request"}]}
Dec 10 18:02:27 172.21.0.1 core[ ]: /harbor/src/lib/http/error.go: ,
github.com/goharbor/harbor/src/lib/http.apiError
Dec 10 18:02:27 172.21.0.1 core[ ]: /harbor/src/lib
```

# Bearer

The situation looks similar to the Basic 

My Config:

![bearer](https://user-images.githubusercontent.com/42082796/101853069-fc11eb80-3b99-11eb-99f5-f16137429df3.png)

Browser Console:

```json
{"errors":[{"code":"UNKNOWN","message":"internal server error"}]}
```

Harbor log:

```txt
Dec 10 18:04:24 172.21.0.1 core[ ]: 2020- - T : : Z [ERROR] [/lib/http/error.go: ]: {"errors":
[{"code":"INTERNAL_SERVER_ERROR","message":"scanner API: ping: scanner controller: ping: v client:
get metadata: send: authorization: Bearer: missing data to authorize request"}]}
Dec 10 18:04:24 172.21.0.1 core[ ]: /harbor/src/lib/http/error.go: ,
github.com/goharbor/harbor/src/lib/http.apiError
Dec 10 18:04:24 172.21.0.1 core[ ]: /harbor/src/lib/http/error.go: ,
github.com/goharbor/harbor/src/lib/http.SendError
Dec 10 18:04:24 172.21.0.1 core[ ]: /harbor/src/com
```

**Comment by heww** [comment]
The target endpoint must implement the metadata API of the [pluggable-scanner-spec](https://github.com/goharbor/pluggable-scanner-spec) otherwise harbor will return 500 when creating a scanner or testing the connection.

**Comment by CNYuYang** [comment]
yes , I was implement the metadata API , now I want implement identity authentication

**Comment by skrech** [comment]
I can confirm that the Bearer Authentication method is not working. When Add button is clicked on the "Add Scanner" dialog the response is as already shown by @CNYuYang:
`2020-12-22T14:26:43Z [ERROR] [/lib/http/error.go:54]: {"errors":[{"code":"UNKNOWN","message":"scanner API: create: api controller: create registration: scanner controller: ping: v1 client: get metadata: send: authorization: Bearer: missing data to authorize request"}]}
`

This error has absolutely nothing to do with implementing the pluggable-scanner-spec, because there is no request made to the adapter. Something is failing within Harbor Core, which prevents it from sending an HTTP request to the adaptor. 

My version of Harbor is: v2.1.1-5f52168e

**Comment by skrech** [comment]
Can we remove the "pending-for-more-info" label? The issue is defined and still present - I have pluggable-scanner-spec implemented and no Authentication other than None is working.

**Comment by heww** [comment]
@skrech Could you share the metadata response of your scanner?

**Comment by skrech** [comment]
Hello @heww,

The response is the same for when using Authorization or not. It looks like:
```
        data = {
            "scanner": {
                "name": "NameOfOurProduct",
                "vendor": "Company Inc.",
                "version": settings.VERSION,
            },
            "capabilities": [
                {
                    "consumes_mime_types": [
                        "application/vnd.docker.distribution.manifest.v2+json"
                    ],
                    "produces_mime_types": [
                        "application/vnd.scanner.adapter.vuln.report.harbor+json; version=1.0"
                    ],

                },
            ],
            # "properties": {
            # },
        }

```

However, I want to stress out that when clicking the Add button there is no request hitting our nginx server, while in the log of the Harbor is logged the following:
```

2021-01-19T08:31:53Z [ERROR] [/lib/http/error.go:54]: {"errors":[{"code":"UNKNOWN","message":"scanner API: create: api controller: create registration: scanner controller: ping: v1 client: get metadata: send: authorization: Bearer: missing data to authorize request"}]}
/harbor/src/core/api/scanners.go:143, github.com/goharbor/harbor/src/core/api.(*ScannerAPI).Create
/usr/local/go/src/reflect/value.go:460, reflect.Value.call
/usr/local/go/src/reflect/value.go:321, reflect.Value.Call
/harbor/src/vendor/github.com/astaxie/beego/router.go:853, github.com/astaxie/beego.(*ControllerRegister).ServeHTTP
/harbor/src/server/middleware/readonly/readonly.go:77, github.com/goharbor/harbor/src/server/middleware/readonly.MiddlewareWithConfig.func1
/harbor/src/server/middleware/middleware.go:57, github.com/goharbor/harbor/src/server/middleware.New.func1.1
/usr/local/go/src/net/http/server.go:2041, net/http.HandlerFunc.ServeHTTP
/harbor/src/server/middleware/security/security.go:63, github.com/goharbor/harbor/src/server/middleware/security.Middleware.func1.1
/usr/local/go/src/net/http/server.go:2041, net/http.HandlerFunc.ServeHTTP
/harbor/src/server/middleware/artifactinfo/artifact_info.go:55, github.com/goharbor/harbor/src/server/middleware/artifactinfo.Middleware.func1.1
/usr/local/go/src/net/http/server.go:2041, net/http.HandlerFunc.ServeHTTP
/harbor/src/server/middleware/transaction/transaction.go:62, github.com/goharbor/harbor/src/server/middleware/transaction.Middleware.func1.1
/harbor/src/lib/orm/orm.go:73, github.com/goharbor/harbor/src/lib/orm.WithTransaction.func1
/harbor/src/server/middleware/transaction/transaction.go:71, github.com/goharbor/harbor/src/server/middleware/transaction.Middleware.func1
/harbor/src/server/middleware/middleware.go:57, github.com/goharbor/harbor/src/server/middleware.New.func1.1
/usr/local/go/src/net/http/server.go:2041, net/http.HandlerFunc.ServeHTTP
/harbor/src/server/middleware/notification/notification.go:31, github.com/goharbor/harbor/src/server/middleware/notification.Middleware.func1
/harbor/src/server/middleware/middleware.go:57, github.com/goharbor/harbor/src/server/middleware.New.func1.1
/usr/local/go/src/net/http/server.go:2041, net/http.HandlerFunc.ServeHTTP
/harbor/src/server/middleware/orm/orm.go:53, github.com/goharbor/harbor/src/server/middleware/orm.MiddlewareWithConfig.func1
/harbor/src/server/middleware/middleware.go:57, github.com/goharbor/harbor/src/server/middleware.New.func1.1
/usr/local/go/src/net/http/server.go:2041, net/http.HandlerFunc.ServeHTTP
/harbor/src/server/middleware/csrf/csrf.go:48, github.com/goharbor/harbor/src/server/middleware/csrf.attach.func1
/usr/local/go/src/net/http/server.go:2041, net/http.HandlerFunc.ServeHTTP
/harbor/src/vendor/github.com/gorilla/csrf/csrf.go:297, github.com/gorilla/csrf.(*csrf).ServeHTTP
/harbor/src/server/middleware/csrf/csrf.go:68, github.com/goharbor/harbor/src/server/middleware/csrf.Middleware.func2
/harbor/src/server/middleware/middleware.go:57, github.com/goharbor/harbor/src/server/middleware.New.func1.1
/usr/local/go/src/net/http/server.go:2041, net/http.HandlerFunc.ServeHTTP
/harbor/src/server/middleware/session/session.go:34, github.com/goharbor/harbor/src/server/middleware/session.Middleware.func1.1
/usr/local/go/src/net/http/server.go:2041, net/http.HandlerFunc.ServeHTTP
/harbor/src/server/middleware/log/log.go:33, github.com/goharbor/harbor/src/server/middleware/log.Middleware.func1
/harbor/src/server/middleware/middleware.go:57, github.com/goharbor/harbor/src/server/middleware.New.func1.1
/usr/local/go/src/net/http/server.go:2041, net/http.HandlerFunc.ServeHTTP
/harbor/src/server/middleware/requestid/requestid.go:37, github.com/goharbor/harbor/src/server/middleware/requestid.Middleware.func1
/harbor/src/server/middleware/middleware.go:57, github.com/goharbor/harbor/src/server/middleware.New.func1.1
/usr/local/go/src/net/http/server.go:2041, net/http.HandlerFunc.ServeHTTP
/usr/local/go/src/net/http/server.go:2836, net/http.serverHandler.ServeHTTP
/usr/local/go/src/net/http/server.go:1924, net/http.(*conn).serve

```

The screenshot right before I click the Add button:
![Screenshot 2021-01-19 at 10 32 04](https://user-images.githubusercontent.com/2196551/105008414-e10c8400-5a41-11eb-8f48-585afbcfd636.png)

Here is an image when I switch to ApiKey (these is no output to the console when using this option):
![Screenshot 2021-01-19 at 10 35 27](https://user-images.githubusercontent.com/2196551/105008639-2af56a00-5a42-11eb-8a2d-366402fff706.png)

**Comment by heww** [comment]
Hi @skrech I had verified that this is a bug and i will fix it in 2.2. May cherry pick it to 2.1.4.
